modules/gdscript/gdscript.h

```diff
@@ -325,6 +329,8 @@ class GDScriptInstance : public ScriptInstance {
 
 	virtual void get_method_list(List<MethodInfo> *p_list) const;
 	virtual bool has_method(const StringName &p_method) const;
+	Callable::CallError call_getter(const GDScript::MemberInfo &p_member_info, const StringName &p_name, Variant &r_ret) const;
+	Callable::CallError call_setter(const GDScript::MemberInfo &p_member_info, const StringName &p_name, const Variant &p_value);
```

Comment:
Note that you have added these methods to `protected` section of `GDScript` , to `public` section of `GDScriptInstance`, and the `MemberInfo` struct is `private`. I see that you did this so that the methods are next to `callp()`, but I think they have a different purpose. In my opinion `call_getter()`/`call_setter()` methods should be private.

Or maybe we should discard them, since there are only 3 calls for 2 declarations and 2 definitions. This is not a lot of code duplication and yet new uses of these methods are not expected, I think.

Reply:
I agree about making the methods private or protected. I'm never really sure when something should be protected versus private, but I'll take your word that private is more suitable in this case.

I'd prefer to keep the helper methods as I think it makes the code more readable and easier to maintain. My general philosophy is that if a comment was needed to explain a block of code , the block should be rolled into a function with a self-explanatory name (big exceptions being for comments that explain some kind of non-obvious implementation or cite a source). That said you have much more experience maintaining big code, so I'm happy to defer to you if you feel strongly.

Reply:
I do not insist on code duplication here. It's just my personal opinion in the particular case, I'd like to know what others think about it.

Regarding the visibility modifiers, I explained the problem: the visibility levels are different. In my opinion it is wrong that a public/protected method has a parameter of a private type (`MemberInfo` struct).